src/redisraft.c

```diff
@@ -247,26 +236,27 @@ static int cmdRaftRequestVote(RedisModuleCtx *ctx, RedisModuleString **argv, int
  *   -MOVED <slot> <addr>:<port> ||
  *   Any standard Redis reply, depending on the command.
  */
-
 static int cmdRaft(RedisModuleCtx *ctx, RedisModuleString **argv, int argc)
 {
     if (argc < 2) {
         RedisModule_WrongArity(ctx);
         return REDISMODULE_OK;
     }
 
-    RaftReq *req = RaftReqInit(ctx, RR_REDISCOMMAND);
-    RaftRedisCommand *cmd = RaftRedisCommandArrayExtend(&req->r.redis.cmds);
+    RaftRedisCommandArray cmds = {0};
+    RaftRedisCommand *cmd = RaftRedisCommandArrayExtend(&cmds);
 
     cmd->argc = argc - 1;
     cmd->argv = RedisModule_Alloc((argc - 1) * sizeof(RedisModuleString *));
 
     int i;
     for (i = 0; i < argc - 1; i++) {
         cmd->argv[i] =  argv[i + 1];
-        RedisModule_RetainString(req->ctx, cmd->argv[i]);
+        RedisModule_RetainString(ctx, cmd->argv[i]);
     }
-    handleRedisCommand(&redis_raft, req);
+    handleRedisCommand(&redis_raft, ctx, &cmds);
+    RaftRedisCommandArrayFree(&cmds);
```

Comment:
Isn't the command array part of the RaftReq at this point and cannot be freed (or at least, the individual strings need to be retained again)?

Reply:
These commands will be moved if needed inside `handleRedisCommand()`. Please see `RaftRedisCommandArrayMove()` calls inside `handleRedisCommand()`. Whether we moved or not, it'll be safe to free this array.